### PR TITLE
implement more admin API functions

### PR DESCRIPTION
**Pull request by hjmodi**
Signed-off-by: Harsh Modi <hmodi@redhat.com>

### CODE REVIEW

**Comment by hjmodi** [comment]
This PR is a little bigger but I created it because it doesn't contain any vitally different functions. Most functions added in this PR are straightforward.

File: src/main/java/org/commonjava/indy/service/tracking/model/ApplicationStatus.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.service.tracking.model;
+
+public enum ApplicationStatus
+{
```
Comment (sswguo) [inline src/main/java/org/commonjava/indy/service/tracking/model/ApplicationStatus.java:19]: Do we need to create this by ourselves for the specific thing ? I remembered that there are some 3rd libraries for this.

Comment (ligangty) [reply src/main/java/org/commonjava/indy/service/tracking/model/ApplicationStatus.java:19]: Agree. I remember that jaxrs provides one: javax.ws.rs.core.Response.Status. I think we can reuse it directly.

File: src/main/java/org/commonjava/indy/service/tracking/jaxrs/ResponseHelper.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    }
+
+    public Response formatResponse( final Throwable error )
+    {
+        return formulateResponse( 0, error, null, false, null );
+    }
+
+    private Response formulateResponse( final int statusCode, final Throwable error, final String message,
+                                        final boolean throwIt, Consumer<ResponseBuilder> builderModifier )
+    {
```
[the remark this answers is not in the document]
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/jaxrs/ResponseHelper.java:91]: changed to call with conversion. :)

File: src/main/java/org/commonjava/indy/service/tracking/model/ApplicationStatus.java (hunk 1 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/model/ApplicationStatus.java]: updated to use ` javax.ws.rs.core.Response.Status`

**Review by ligangty — COMMENTED** [review]
Generally looks good with some little questions.

**Review by ligangty — APPROVED** [review]
LGTM

**Pull request merged**